ts/sagas/instabug.ts

```diff
@@ -20,8 +30,38 @@ const onNewReplyReceived = () =>
     });
   });
 
-function* watchInstabugSaga(): IterableIterator<Effect> {
+function initializeInstabugSDKDismissalListener(): EventChannel<unknown> {
```

Comment:
You could avoid to use eventChannel by using a promise and fork a watcher
What do you think about this attempt?

96f00024bf5dd177266a745915d61ccdaa8c9a6a

Reply:
To be honest, I find that to be more raw, IMHO we should use saga helpers without reinventing the wheel, unless you're aware of `eventChannel` performance problems. In particular, `eventChannel` should always be used when we're dealing with event-driven programming. We should use `while (true)` only when we can't do without it, i.e. when making our own helpers :)

Reply:
Let me know if I'm going against any codebase-related rules.